### PR TITLE
Don't use multiprocessing when -p 1 to simplify and ease debugging

### PR DESCRIPTION
**Pull request by yesimon**
Add mkdir_p utility function

### CODE REVIEW

File: ncbi_genome_download/core.py [hunk 1]
```diff
@@ -227,28 +227,33 @@ def download(**kwargs):
...
+                                 species_taxid, taxid, human_readable, refseq_category, table):
+                yield job
 
+    try:
+        if parallel == 1:
+            for job in download_jobs_groups():
+                worker(job)
+        else:
+            pool = Pool(processes=parallel)
+            jobs = pool.map_async(worker, download_jobs)
```
Comment (kblin) [inline ncbi_genome_download/core.py]: I don't think this will work anymore. Have you tried parallel downloads with your changes? This is always a bit tricky to see looking at diffs, but I'm missing where `download_jobs` is defined now.

File: ncbi_genome_download/core.py [hunk 2]
```diff
@@ -366,33 +370,33 @@ def download_entry(entry, section, domain, output, file_format, human_readable,
...
             logging.error(err)
 
-    return download_jobs
 # pylint: enable=too-many-arguments
 
-
-def create_dir(entry, section, domain, output):
-    """Create the output directory for the entry if needed"""
-    full_output_dir = os.path.join(output, section, domain, entry['assembly_accession'])
+def mkdir_p(path):
```
Comment (kblin) [inline ncbi_genome_download/core.py]: This seems to be an unrelated change. One I'd happily consider in a separate PR, though.

File: ncbi_genome_download/core.py [hunk 3]
```diff
@@ -227,28 +227,33 @@ def download(**kwargs):
...
-        pool = Pool(processes=parallel)
-        jobs = pool.map_async(worker, download_jobs)
-        try:
-            # 0xFFFF is just "a really long time"
-            jobs.get(0xFFFF)
-        except KeyboardInterrupt:  # pragma: no cover
-            # TODO: Actually test this once I figure out how to do this in py.test
-            logging.error("Interrupted by user")
-            return 1
+    def download_jobs_groups():
```
Comment (kblin) [inline ncbi_genome_download/core.py]: Is there a particular reason you're switching to a generator here?

**Comment by yesimon** [comment]
If you're doing major change then no need to merge - I just hope there will be a simpler serial code execution path (especially with 1 thread).

File: ncbi_genome_download/core.py (hunk 1 above)
Comment (yesimon) [reply ncbi_genome_download/core.py]: Yeah I pulled this out of a fork with more changes and it might have gotten lost.

File: ncbi_genome_download/core.py (hunk 3 above)
Comment (yesimon) [reply ncbi_genome_download/core.py]: I feel like the generator fits a bit better with the single threaded serially processing= mode (better than lists of jobs).

**Comment by kblin** [comment]
I finally got around to push a patch e9d792b to single-thread `-p 1` runs.

**Comment by kblin** [comment]
Er, never mind, didn't push yet, the sha1 will likely change.

**Comment by kblin** [comment]
There we go: 068dd74 is the one.

**Comment by kblin** [comment]
Closing the pull request, this feature will be in the 0.2.6 release.
